From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spottedleaf <spottedleaf@spottedleaf.dev>
Date: Fri, 28 Feb 2020 19:38:19 -0800
Subject: [PATCH] Ensure loadCallback() and unloadCallback() exceptions are
 printed

By default they are not because they are wrapped in thenAccept.

diff --git a/src/main/java/net/minecraft/server/PlayerChunk.java b/src/main/java/net/minecraft/server/PlayerChunk.java
index 1db523546ce7d633f3b18f4c01555812330aa627..85f32fd4d3f384fdd2fd158165fb7c8480c3c047 100644
--- a/src/main/java/net/minecraft/server/PlayerChunk.java
+++ b/src/main/java/net/minecraft/server/PlayerChunk.java
@@ -438,7 +438,10 @@ public class PlayerChunk {
                     chunk.setNeedsSaving(true);
                     chunk.unloadCallback();
                 });
-            }, playerchunkmap.callbackExecutor);
+            }, playerchunkmap.callbackExecutor).exceptionally((Throwable thr) -> {  // Tuinity start - ensure this gets logged - fuck you completablefuture api
+                thr.printStackTrace();
+                return null;
+            }); // Tuinity - end
 
             // Run callback right away if the future was already done
             playerchunkmap.callbackExecutor.run();
@@ -600,7 +603,10 @@ public class PlayerChunk {
                     Chunk chunk = (Chunk) chunkAccess;
                     chunk.loadCallback();
                 });
-            }, playerchunkmap.callbackExecutor);
+            }, playerchunkmap.callbackExecutor).exceptionally((Throwable thr) -> {  // Tuinity start - ensure this gets logged - fuck you completablefuture api
+                thr.printStackTrace();
+                return null;
+            }); // Tuinity - end
 
             // Run callback right away if the future was already done
             playerchunkmap.callbackExecutor.run();
